source/slang/source-loc.cpp

```diff
@@ -332,6 +332,19 @@ void SourceManager::initialize(
     m_nextLoc = m_startLoc;
 }
 
+SourceManager::~SourceManager()
+{
+    for (auto item : m_sourceViews)
```

Comment:
I suppose we could have made the arrays use `std::unique_ptr` to avoid the explicit destructor? Any opinion on that?

Reply:
I think that using a smart pointer like unique_ptr would be superior. Presumably in our case it would be a creation of our own Slang::UniquePtr tho. I think it would be a good thing to have.